### ISSUE TITLE
Figure out how to include next encounter steps to disambiguate more seeds

### ISSUE THREAD

**Issue by Apjjm** [Apjjm/EncounterTool#2]
The baron manip chart on the wiki doesn't just look at the first formation, it also looks at the number of steps for the next encounter. This allows it to find a couple of extra cells that give good encounters than just the formation data used here does for mac giants at baron (so probably other encounters too).

How do we want to render this nicely to fit in a cell here?
 - Each cell is either (A) formations based, (B) next step based (if steps < 10) (C) just `? X`
 - Include steps needed for next encounter, if encounters > 1 in data from the walker?
 - Do a 2nd group by pass and group by next steps too.
   - If all good encounters succeed in < X include them
   - If all bad encounters fail in < X and all good encounters > max(bad) then include it?
 - When deciding which is better, prefer the formation based?

**Comment by Apjjm** [comment]
We now store and log the how many steps there are until the next encounter.

Actually showing this logic in the squares gets complicated quickly because:
 - Sometimes it is formation dependent
 - Sometimes we want to use step counting to say "this encounter is good"
 - Sometimes we want to use step counting to say "these encounters are bad"
 - Sometimes the step counting will give fewer encounters to go through, than just picking the common encounter a bit later. In practice this may or may not be faster for the runner (and just because encounter # is lower it could still mean more steps taken to get the desired encounter in the goal map!)

I think a general approach for this would want to generate 2 subclasses of StepData objects - a "step count" based data and a "formation" based data. We would likely want to split apart MapWalker to just do the map walking and encounter list generation, and then a factory for each set of step data to calculate them in isolation for the most part.

Finally, we could then do a combination pass for the step data objects where we decide (through some, probably quite complex) comparator which of the two datas to actually pick for the step.

The rendered step object would need to have a strategy/function for handling each of the different types of step.

This approach still wouldn't be entirely general - we couldn't do step counting for 1 formation and something different for a different formation. It also would likely mean that formations that target multiple different rooms might not always merge optimially on a given step if two different rooms use different strategies, depending on when we merge stuff.

In short - this might be something easier to just add in post, at the map makers descretion, when putting the image together for the few cells where it actually matters. The data is all available to see in a CSV and is easy enough to sort by step.

**Comment by Apjjm** [comment]
added a first attempt at this now

made some slightly different tradeoffs to the ones in the stepcharts on the wikis (mainly avoiding counting 50+ steps and preferring using the formation data - this is why this tool exists in the first place!)
